src/algorithms/fardetectors/FarDetectorTrackerCluster.cc

```diff
@@ -127,22 +129,20 @@ void  FarDetectorTrackerCluster::ClusterHits(const edm4eic::TrackerHitCollection
       // Removes current hit from remaining found cluster hits
       clusterList.erase(clusterList.begin());
 
-      // Adds raw hit to TrackerHit contribution
-      clusterHits.push_back(inputHits[index]);
-
-      // Energy
-      auto hitE = e[index];
-      esum += hitE;
       // TODO - See if now a single detector element is expected a better function is available.
       auto pos = m_seg->position(id[index]);
 
       // Weighted position
-      float weight = hitE; // TODO - Calculate appropriate weighting based on sensor charge sharing
+      float weight = e[index]; // TODO - Calculate appropriate weighting based on sensor charge sharing
       weightSum += weight;
       localPos += pos * weight;
 
       // Time
       clusterT.push_back(t[index]);
+
+      // Adds hit and weight to Measurement2D contribution
+      cluster.addToHits(inputHits[index]);
+      cluster.addToWeights(weight);
```

Comment:
For clustering, and for MC associations in general we lean towards using normalized weights (fractions). Is there a reason to not do that here?

Reply:
No particular motivation at the moment, happy to swap over.